plugins/source/aws/codegen/recipes/base.go

```diff
@@ -82,6 +83,26 @@ func awsNameTransformer(f reflect.StructField) (string, error) {
 	return c.ToSnake(f.Name), nil
 }
 
+func awsResolverTransformer(f reflect.StructField, path string) (string, error) {
+	if f.Type.String() == "[]types.Tag" {
+		if path == "Tags" {
+			return "client.ResolveTags", nil
+		}
+		return `client.ResolveTagField("` + path + `")`, nil
+	}
+
+	if path == "Tags" || path == "TagSet" {
+		switch f.Type {
+		case reflect.TypeOf(map[string]string{}), reflect.TypeOf(map[string]*string{}), reflect.TypeOf(map[string]interface{}{}), reflect.TypeOf([]types.TagDescription{}):
+			// valid tag types
+		default:
+			return "", fmt.Errorf("%q field is not of type []types.Tag or acceptable map: %s", path, f.Type.String())
+		}
+	}
+
+	return `schema.PathResolver("` + path + `")`, nil
```

Comment:
Maybe let's use `codegen.DefaultResolverTransformer` as the default here, so we automatically benefit from future SDK improvements of the default?

Reply:
Good idea, fixed in [`a953d78` (#3330)](https://github.com/cloudquery/cloudquery/pull/3330/commits/a953d78f26cdb5a4d433926a5e9112a18e58c112)